### PR TITLE
Incorporate feedback from the Ethical ML workshops

### PR DESCRIPTION
**Pull request by anssiko**
- Update Operationalization: Putting the Principles into Practice
- Add Register of Risks and Mitigations

***
<a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/ethical-webmachinelearning/pull/20.html" title="Last updated on Jun 2, 2022, 1:31 PM UTC (a272f56)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/ethical-webmachinelearning/20/25a8ba5...a272f56.html" title="Last updated on Jun 2, 2022, 1:31 PM UTC (a272f56)">Diff</a>

### CODE REVIEW

**Comment by anssiko** [comment]
Gentle ping, PTAL @dontcallmedom :-)

Some areas to focus on, mostly presentational:

- Proposals welcome on how to better represent Risks -> Possible Mitigations mapping. Maybe with some CSS tricks we could use a two-column layout for Risks and Possible Mitigations side-by-side (when screen estate allows)? Someone with proper CSS-fu would need to help with that. Or we could consider that future work.

- "This bullet intentionally left blank" in Possible Mitigations. I was thinking of turning these empty bullets into links that open a new GH issue to encourage contributions.

Comments:

- I'd prefer to keep the source format as simple as possible and let CSS handle presentational aspects, if possible. Complex tables are tedious to edit.

- To help editing the lists and correlate Risks with Possible Mitigations in source, maybe instead of:
```
1.

1.

1.
```

We should use:

```
1.

2.

3.
```

File: index.bs [hunk 1]
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
...
+At its simplest, the core approach is to take known areas of concern - as laid out by the principles and guidance - and for each of them think through what could go wrong (consequences/risks/harms), and what measures could be put in place to prevent that or minimise impact (mitigations).
 
+So for example, if we consider the principle of “Fairness and non-dicrimination”, a risk might be that biases in training data lead to model predictions that are less accurate for particular groups, resulting in real-world harms (e.g. denial of services). A mitigation might be to test the training data and model properly for bias, or indeed it might be to not make access to essential services dependent on ML systems.
+
+To provide a more structured approach to this, we have developed a workshop format which you can adapt and use.
```
Comment (dontcallmedom) [inline index.bs]:
```suggestion
To provide a more structured approach to this, we have developed a workshop format that can be adapted and used to help identify such risks & mitigations.
```

File: index.bs [hunk 2]
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
...
+
+The workshop is in two parts (please make a copies for your own use):
```
Comment (dontcallmedom) [inline index.bs]:
```suggestion
The workshop is articulated around two documents that are expected to be completed interactively by participants:
```

File: index.bs [hunk 3]
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
...
+
+Part One - [Ethical thinking workshop](https://docs.google.com/document/d/1f_PcByjW8-zXbYWeEyOl-RpZ3SKapm_MWSaNZ9ZOi4c/edit?usp=sharing) - is about using the principles to generate and prioritize potential risks
+
+Part Two - [Ethical Risk Canvas](https://docs.google.com/document/d/1hTQnpWC5KC4qIJB9-Kkd46yMVgDuTCQlA3MffqtUbCI/edit?usp=sharing) - is about digging deeper into specific risks and thinking about who they might impact and how best to mitigate them.
+
```
Comment (dontcallmedom) [inline index.bs]: we should probably post a PDF copy of these documents somewhere if only for archival sake.

File: index.bs (hunk 1 above)
Comment (dontcallmedom) [reply index.bs]: I wonder if the details about the workshop (until "it's worth noting" below) should be moved to an appendix - while very useful, it feels a bit out of place in the flow of the document.

File: index.bs [hunk 4]
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
...
+As this suggests, referring to the higher level values such as fundamental human rights and freedoms might help. Fairness is often a useful guide to balancing competing interests and values. Bear in mind W3C’s [priority of constituencies](https://www.w3.org/TR/design-principles/#priority-of-constituencies).
+
+And process is often as important as outcome. The principles in this document don’t just cover what issues you should think about (such as bias) but also importantly suggest how an ethical process should be run - ideally it should involve diverse participants, consult affected stakeholders, be open, democratic and transparent (so keep a record of the process and ideally make the results publicly available) and open to contestation.
+
+# Register of Risks and Mitigations
+
+Note: The risk register is work in progress and welcomes further review and tidying up.
```
Comment (dontcallmedom) [inline index.bs:479]: I'm happy to leave this for a future pull request, but I think there is indeed substantive work to harmonize the phrasing of the risks, with possibly merging / sorting some of them as well.

File: index.bs [hunk 5]
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
...
+
+1. One example is how geographic pricing is employed by companies like Amazon – e.g. depending on where your IP address is located to or depending on your device type, different prices are presented to shoppers – it’s a question if this is unethical or unlawful, but it is something that is happening and also begs to question whether or not this is something we want to speak to → taken further you can also imagine reducing service based on geography or hardware or other factors in a way that is automated through ML systems
+
+### Possible Mitigations
+
+1. Browser-assisted mechanisms to find out about the limitations and performance characteristics of ML models used in a Web app. This could build on an approach published in Model Cards for Model Reporting where making this report machine-discoverable would allow for the web browser to offer a more integrated user experience. Another transparency tool is the [Open Ethics Transparency Protocol](https://github.com/webmachinelearning/ethical-webmachinelearning/issues/6).
```
Comment (dontcallmedom) [inline index.bs]: In terms of mapping mitigations to risks, I would use a naming scheme for risks (e.g. FND-R1 for risk 1 under **F**airness and **N**on-**D**iscrimination), and then references these names from the mitigations that address them.

Comment (anssiko) [reply index.bs]: Done in https://github.com/webmachinelearning/ethical-webmachinelearning/pull/20/commits/b00171e56c8d40f936040d19c877b5fccc972aba

File: index.bs (hunk 1 above)
Comment (anssiko) [reply index.bs]: Moved to appendix. Text changes integrated into https://github.com/webmachinelearning/ethical-webmachinelearning/pull/20/commits/13d34dc490bdea3ce0341c4b39f26eb2000c206e

File: index.bs (hunk 2 above)
Comment (anssiko) [reply index.bs]: Integrated into https://github.com/webmachinelearning/ethical-webmachinelearning/pull/20/commits/13d34dc490bdea3ce0341c4b39f26eb2000c206e

File: index.bs (hunk 3 above)
Comment (anssiko) [reply index.bs]: Opened https://github.com/webmachinelearning/ethical-webmachinelearning/issues/21

**Comment by anssiko** [comment]
Thanks @dontcallmedom for your comments, PR updated.

**Comment by anssiko** [comment]
Gentle ping, PTAL @dontcallmedom.

Your feedback has been addressed. I'd like to merge this after your approval so the WG can advance with its W3C Note publication plan.

**Review by dontcallmedom — APPROVED** [review]

**Comment by anssiko** [comment]
TY @dontcallmedom!

**Pull request merged**